src/api/tokens.rs

```diff
@@ -0,0 +1,54 @@
+use actix::prelude::*;
+use actix_web::web::{Data, Json};
+use actix_web::{HttpRequest, HttpResponse, Result};
+use futures3::TryFutureExt;
+use serde::Deserialize;
+
+use crate::db::Db;
+use crate::errors::ApiError;
+use crate::tokens::{ClaimsScope, ClaimsValidator};
+
+#[derive(Deserialize)]
+pub struct TokenArgs {
+    token_ids: Vec<String>,
+}
+
+pub fn get_tokens(
+    args: Json<TokenArgs>,
+    db: Data<Db>,
+    req: HttpRequest,
+) -> impl Future<Item = HttpResponse, Error = ApiError> {
```

Comment:
Any reason to not use the `async` keyword? It does more than just being eye candy.

Reply:
We are using a very old version of Actix that uses futures 0.1, so the extra functions convert from std futures to the old ones with `.compat()`.